src/test/pybind/test_rbd.py

```diff
@@ -854,6 +854,12 @@ def test_metadata(self):
             metadata = list(self.image.metadata_list())
             eq(len(metadata), N - i - 1)
 
+    def test_watchers_list(self):
+        watchers = list(self.image.watchers_list())
+        # The image is open, in r/w mode, from a previous test, so expect there
```

Comment:
Nit: the image is opened as part of the test setup (https://github.com/ceph/ceph/pull/19188/commits/1e5c0daaca778694e6f4dfd73f59e4095cedd27e#diff-1e9029bdf63884f2975c4a2bd8fab90aR300) and not part of a previous test

Reply:
Ah, good catch. I'm used to the go bindings, where creating an `Image` struct is a separate step from opening it, so I assumed the `Image` creation in the setup was not what did the open. I've updated the comment.

Trying to run the tests fails for me. I think it's looking for a running ceph cluster that I don't have configured locally:

```console
% nosetests -v test_rbd -e '.*many_snaps'
unable to get monitor info from DNS SRV with service name: ceph-mon
no monitors specified to connect to.
ERROR

======================================================================
ERROR: test suite for <module 'test_rbd' from '/home/awg/ceph/src/test/pybind/test_rbd.pyc'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/usr/lib/python2.7/dist-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/awg/ceph/src/test/pybind/test_rbd.py", line 39, in setup_module
    rados.connect()
  File "rados.pyx", line 878, in rados.Rados.connect (/home/awg/ceph/build/src/pybind/rados/rados.c:11665)
ObjectNotFound: [errno 2] error connecting to the cluster

----------------------------------------------------------------------
Ran 0 tests in 0.187s

FAILED (errors=1)```

If you can give the python bits a test that would be great.